src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecProc.java

```diff
@@ -86,7 +92,24 @@ public int join() throws IOException, InterruptedException {
             LOGGER.log(Level.FINEST, "Waiting for websocket to close on command finish ({0})", finished);
             finished.await();
             LOGGER.log(Level.FINEST, "Command is finished ({0})", finished);
-            return watch.exitCode().join();
+
+            CompletableFuture<Integer> exitCodeFuture = watch.exitCode();
+            Integer exitCode = exitCodeFuture.get();
+
+            if (exitCode == null) {
+                LOGGER.log(Level.FINEST, "Watcher return 'null' instead of exitCode");
```

Comment:
I don't think it is useful to mention `null` in a user facing log message if we can better diagnose it,
```suggestion
                LOGGER.log(Level.FINEST, "The container exec watch was closed before it could obtain an exit code from the process.");
```
The same message should be used as well for what gets sent to the print stream.